src/util.c

```diff
@@ -17,3 +17,84 @@ long parse_int(char *str) {
 
     return parsed;
 }
+
+unsigned long parse_buttons_numbers(char *str,
```

Comment:
By not supporting ranges this should become much less complex by using something like `strtok`.